src/libratbag.c

```diff
@@ -1273,6 +1299,93 @@ ratbag_button_unref(struct ratbag_button *button)
 	return NULL;
 }
 
+LIBRATBAG_EXPORT enum ratbag_led_mode
+ratbag_led_get_mode(struct ratbag_led *led)
+{
+	return led->mode;
+}
+
+LIBRATBAG_EXPORT struct ratbag_color
+ratbag_led_get_color(struct ratbag_led *led)
+{
+	return led->color;
+}
+
+LIBRATBAG_EXPORT int
+ratbag_led_get_rate(struct ratbag_led *led)
+{
+	return led->rate;
+}
+
+LIBRATBAG_EXPORT int
+ratbag_led_get_bright(struct ratbag_led *led)
+{
+	return led->bright;
+}
+
+LIBRATBAG_EXPORT enum ratbag_error_code
+ratbag_led_set_mode(struct ratbag_led *led, enum ratbag_led_mode mode)
+{
+	/* if (!ratbag_device_has_capability(button->profile->device, */
+  /*                                   RATBAG_DEVICE_CAP_BUTTON_KEY)) */
+	/* 	return RATBAG_ERROR_CAPABILITY; */
+
+	led->mode = mode;
+	led->dirty = true;
+	led->profile->dirty = true;
+	return RATBAG_SUCCESS;
+}
+
+LIBRATBAG_EXPORT enum ratbag_error_code
+ratbag_led_set_color(struct ratbag_led *led, struct ratbag_color color)
+{
+	/* if (!ratbag_device_has_capability(button->profile->device, */
+  /*                                   RATBAG_DEVICE_CAP_BUTTON_KEY)) */
+	/* 	return RATBAG_ERROR_CAPABILITY; */
+
+	led->color = color;
+	led->dirty = true;
+	led->profile->dirty = true;
+	return RATBAG_SUCCESS;
+}
+
+LIBRATBAG_EXPORT enum ratbag_error_code
+ratbag_led_set_rate(struct ratbag_led *led, unsigned int rate) {
+	led->rate = rate;
+	led->dirty = true;
+	led->profile->dirty = true;
+	return RATBAG_SUCCESS;
+}
+
+LIBRATBAG_EXPORT enum ratbag_error_code
+ratbag_led_set_bright(struct ratbag_led *led, unsigned int bright) {
+	led->bright = bright;
+	led->dirty = true;
+	led->profile->dirty = true;
+	return RATBAG_SUCCESS;
+}
+
+LIBRATBAG_EXPORT struct ratbag_led*
+ratbag_profile_get_led(struct ratbag_profile *profile,
+                       enum ratbag_led_type type)
+{
+	struct ratbag_device *device = profile->device;
+
+	switch (type) {
+	case RATBAG_LED_TYPE_UNKNOWN:
+		return NULL;
+	case RATBAG_LED_TYPE_LOGO:
+		return profile->logo_led;
+	case RATBAG_LED_TYPE_SIDE:
+		return profile->side_led;
+	}
```

Comment:
yeah, looking at this now after the comment above: we can just make this an array and loop through it until we find the type.